cmd/update/project.go

```diff
@@ -48,45 +42,106 @@ Incorrect usage when passing unknown-project:
 
  flytectl update project unknown-project --archiveProject
 
-Incorrect usage when passing valid project using -p option:
+project ID is required flag
 
 ::
 
  flytectl update project unknown-project --archiveProject -p known-project
 
+Update projects.(project/projects can be used interchangeably in these commands)
+
+::
+
+ flytectl update project --id flytesnacks --description "flytesnacks description"  --labels app=flyte
+
+Update a project by definition file. Note: The name shouldn't contain any whitespace characters.
+::
+
+ flytectl update project --file project.yaml 
+
+.. code-block:: yaml
+
+    id: "project-unique-id"
+    name: "Name"
+    labels:
+     app: flyte
+    description: "Some description for the project"
+
+Update a project state by definition file. Note: The name shouldn't contain any whitespace characters.
+::
+
+ flytectl update project --file project.yaml  --archiveProject
+
+.. code-block:: yaml
+
+    id: "project-unique-id"
+    name: "Name"
+    labels:
+     app: flyte
+    description: "Some description for the project"
+
 Usage
 `
 )
 
-var DefaultProjectConfig = &ProjectConfig{}
-
 func updateProjectsFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
-	id := config.GetConfig().Project
-	if id == "" {
-		fmt.Printf(clierrors.ErrProjectNotPassed)
-		return nil
+	projectSpec := project.Definition{}
+	if len(project.DefaultProjectConfig.File) > 0 {
+		yamlFile, err := ioutil.ReadFile(project.DefaultProjectConfig.File)
+		if err != nil {
+			return err
+		}
+		err = yaml.Unmarshal(yamlFile, &projectSpec)
+		if err != nil {
+			return err
+		}
+	} else {
+		projectSpec.ID = project.DefaultProjectConfig.ID
+		projectSpec.Description = project.DefaultProjectConfig.Description
+		projectSpec.Name = project.DefaultProjectConfig.Name
+		projectSpec.Labels = project.DefaultProjectConfig.Labels
+	}
```

Comment:
Move this to getProjectSpecFromConfig

Reply:
Created a common method for reading the config in util